src/client/Microsoft.Identity.Client/Internal/Requests/Silent/BrokerSilentStrategy.cs

```diff
@@ -63,6 +63,7 @@ private async Task<MsalTokenResponse> SendTokenRequestToBrokerAsync()
             }
 
             ValidateResponseFromBroker(msalTokenResponse);
+            Metrics.IncrementTotalAccessTokensFromIdP();
```

Comment:
The android broker may return a result from its internal cache rather than the IDP. I am not sure if there is anything in the broker response that indicates whether the token came from the IDP or the internal cache. Maybe worth a check. Otherwise, you can consider all tokens from the broker the idp? Does WAM have something similar?

Reply:
I think WAM is the same, we don't know if it's from IdP or cache. Since we have `TokenSource.Broker`, I could add separate token count for that for consistency?

Reply:
#Resolved